lib/gwdb-legacy/outbase.ml

```diff
@@ -320,79 +434,14 @@ let output ?(save_mem = false) ?(tasks = []) base =
   load_descends_array base;
   load_strings_array base;
   let strings_data = StringData.of_base base in
-  let lowercase_surname_data_file =
-    Filename.concat base.data.bdir Database.lowercase_surname_data_file
-  in
-  let lowercase_surname_index_file =
-    Filename.concat base.data.bdir Database.lowercase_surname_index_file
-  in
-  let lowercase_first_name_data_file =
-    Filename.concat base.data.bdir Database.lowercase_first_name_data_file
-  in
-  let lowercase_first_name_index_file =
-    Filename.concat base.data.bdir Database.lowercase_first_name_index_file
+  let commit_lowercase_first_name_index =
+    generate_lowercase_first_name_index ~strings_data base
   in
-  trace "create first name lower index";
-  output_first_name_lower_index strings_data base tmp_fnames_lower_inx
-    tmp_fnames_lower_dat;
-  trace "create surname lower index";
-  output_surname_lower_index strings_data base tmp_snames_lower_inx
-    tmp_snames_lower_dat;
-  let oc = Secure.open_out_bin tmp_base in
-  let oc_acc = Secure.open_out_bin tmp_base_acc in
-  let output_array arrname arr =
-    let bpos = pos_out oc in
-    if !verbose then Printf.eprintf "*** saving %s array\n" arrname;
-    flush stderr;
-    arr.output_array oc;
-    let epos = Iovalue.output_array_access oc_acc arr.get arr.len bpos in
-    if epos <> pos_out oc then count_error epos (pos_out oc)
+  let commit_lowercase_surname_index =
+    generate_lowercase_surname_index ~strings_data base
   in
+  let commit_base = generate_base base in
```

Comment:
this is now out of the try, I think generating the files should remain inside the try with to avoid partial sync of the base.